[*] Running ...
[*] Dataset loaded: D:\MCTS-AHD-master\problems\tsp_constructive\dataset\train50_dataset.npy with 64 instances.
D:\MCTS-AHD-master\problems\tsp_constructive\gpt.py:11: RuntimeWarning: invalid value encountered in scalar divide
  bias_factor = 0.1 + 0.4 * (1 - remaining_distance / max(distance_matrix[current_node][destination_node] for dest in unvisited_nodes if dest != current_node))
Traceback (most recent call last):
  File "D:\MCTS-AHD-master\problems\tsp_constructive\eval.py", line 82, in <module>
    obj = eval_heuristic(node_positions[i])
  File "D:\MCTS-AHD-master\problems\tsp_constructive\eval.py", line 52, in eval_heuristic
    raise KeyError(f"Node {next_node} is already visited.")
KeyError: 'Node None is already visited.'
